extensions/notebook/src/book/bookTrustManager.ts

```diff
@@ -0,0 +1,161 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as constants from './../common/constants';
+import { BookTreeItem } from './bookTreeItem';
+import { BookModel } from './bookModel';
+
+export interface IBookTrustManager {
+	isNotebookTrustedByDefault(notebookUri: string): boolean;
+	setBookAsTrusted(bookRootPath: string): boolean;
+	setBookAsUntrusted(bookRootPath: string): boolean;
+}
+
+export interface IBookTrustManagerWorkspaceDetails {
+	workspaceFolders: vscode.WorkspaceFolder[];
+	getConfiguration(section?: string | undefined): vscode.WorkspaceConfiguration;
+}
+
+enum TrustBookOperation {
+	Add,
+	Remove
+}
+
+export class BookTrustManager implements IBookTrustManager {
+
+	private static notebookConfiguration: string = constants.notebookConfigKey;
+	private static notebookTrustedBooksConfiguration: string = constants.trustedBooksConfigKey;
+	private trustedLocalBooks: string[] = [];
+
+	constructor(private books: BookModel[], private workspaceDetails?: IBookTrustManagerWorkspaceDetails) {
+		if (!workspaceDetails) {
+			this.workspaceDetails = {
+				get getConfiguration() {
+					return vscode.workspace.getConfiguration;
+				},
+				get workspaceFolders() {
+					return [].concat(vscode.workspace.workspaceFolders || []);
+				}
+			};
+		}
+	}
+
+	isNotebookTrustedByDefault(notebookUri: string): boolean {
+		let allBooks: BookTreeItem[] = this.getAllBooks();
+		let trustableBookPaths = this.getTrustableBookPaths();
+		let trustableBooks: BookTreeItem[] = allBooks
+			.filter(bookItem => trustableBookPaths.some(trustableBookPath => trustableBookPath === path.join(bookItem.book.root, path.sep)));
+		let isNotebookTrusted = this.isNotebookPartOfBooks(notebookUri, trustableBooks);
+
+		return isNotebookTrusted;
+	}
+
+	isNotebookPartOfBooks(notebookUri: string, books: BookTreeItem[]) {
+		let normalizedNotebookUri = path.normalize(notebookUri);
+		let isPartOfNotebook: boolean = false;
+		let trustedBookItem: BookTreeItem = books.find(bookItem => normalizedNotebookUri.startsWith(path.join(bookItem.book.root, path.sep)));
+
+		if (trustedBookItem) {
+			let fullBookBaseUriWithContent = path.join(trustedBookItem.book.root, 'content', path.sep);
+			let requestingNotebookFormattedUri = normalizedNotebookUri.replace(fullBookBaseUriWithContent, '').replace('.ipynb', '');
+			let notebookInTOC = trustedBookItem.book.tableOfContents.sections.find(jupyterSection => {
```

Comment:
This kind of logic (determining if a particular Notebook is part of a book) feels like it really needs to be pulled into a more generic place like the BookModel

Reply:
Moved it out to `BookTreeItem`. For me it does not make sense to put it in the `BookModel` as it seems to more appropriately represent a BookModelNavigator/View. This entity has multiple books and so no unique identifier for books (perhaps maybe by root, but that feels more like a pseudo identifier) can be supplied to specify the search. Ultimately, if we wanted in the `BoodModel` we could aggregate if from the individual `BookTreeItem`s we have here.